dev/io.openliberty.concurrent.internal/src/io/openliberty/concurrent/internal/processor/ContextServiceDefinitionBinding.java

```diff
@@ -110,7 +110,7 @@ public void merge(ContextServiceDefinition annotation, Class<?> instanceClass, M
 
         unchanged = mergeAnnotationValue(unchanged, XMLunchanged, annotation.unchanged(), KEY_UNCHANGED, DEFAULT_UNCHANGED);
 
-        qualifier = mergeAnnotationValue(qualifier, XMLqualifer, annotation.qualifiers(), KEY_QUALIFIER, DEFAULT_QUALIFIER);
+        qualifiers = mergeAnnotationValue(qualifiers, XMLqualifers, annotation.qualifiers(), KEY_QUALIFIERS, DEFAULT_QUALIFIERS);
```

Comment:
I don't think this will behave correctly for a more complex type like `Class[]`.
mergeAnnotationValue in InjectionBinding does `newValue.equals(oldValue)` which will be an instance comparison for arrays.  It looks like there is a special variant, mergeAnnotationProperties, to handle merging for the properties, and it is even more complex in that it appears to combine individual properties from one list into the other rather than just overwriting.  Should the qualifiers behave similarly where the merge is a combination of the annotation values and `<qualifier>` elements?  Or should the set of the latter replace the former?

This same comment will apply to all 4 resource types.

Reply:
Haven't done any builds on this yet and still waiting on reviews.  I think I'll tackle this now.

Reply:
@njr-11 Resolved this by adding an extra check when we merge XML values under this commit [`b181523` (#27148)](https://github.com/OpenLiberty/open-liberty/pull/27148/commits/b181523bd1b1263ccf3ce1a2894b7d1a1b07f7c0)